yaksh/models.py

```diff
@@ -2062,6 +2063,52 @@ def get_question_answers(self):
                 }]
         return q_a
 
+    def get_diff_of_consecutive_submissions_for_all_code_question(self):
+        """
+            Return diff of consecutive submissions for all questions
+        """
+        q_a = {} # key : question, value : all submitted answers for this question
+        for answer in self.answers.all():
```

Comment:
You already know the questions from ```AnswerPaper.questions```, You can use dict comprehension and smarter queries, like so:

```q_a = {qtn: self.answers.filter(question=qtn) for qtn in self.questions} # q_a variable name needs to be better```

or a similar kind of query to avoid multiple if-else conditions and list append operations

Reply:
done